sql/analyzer/pushdown.go

```diff
@@ -700,3 +705,85 @@ func (es exprSlice) String() string {
 	}
 	return sb.String()
 }
+
+// stripDecorations removes *plan.DecoratedNode that wrap plan.ResolvedTable instances.
+// Without this step, some prepared statement reanalysis rules fail to identify
+// filter-table relationships.
+func stripDecorations(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope) (sql.Node, error) {
+	return plan.TransformUp(node, func(node sql.Node) (sql.Node, error) {
+		switch n := node.(type) {
+		case *plan.DecoratedNode:
+			return n.Child, nil
+		default:
+			return node, nil
+		}
+	})
+
+}
+
+// unresolveTables is a quick and dirty way to make prepared statement reanalysis
+// resolve the most up-to-date table roots while preserving projections folded into
+// table scans.
+func unresolveTables(ctx *sql.Context, a *Analyzer, node sql.Node, scope *Scope) (sql.Node, error) {
+	return plan.TransformUpCtx(node, nil, func(c plan.TransformContext) (sql.Node, error) {
+		switch n := c.Node.(type) {
+		case *plan.ResolvedTable:
+			var table string
+			if n.Table != nil {
+				table = n.Table.Name()
+			}
+			var db string
+			if n.Database != nil {
+				db = n.Database.Name()
+			}
+
+			rt, err := resolveTable(ctx, plan.NewUnresolvedTable(table, db), a)
```

Comment:
If you're resolving the table here, why do you also do it in an analyzer rule?

Reply:
can bindvars be table names? or are they strictly literals?